src/api/routers/participantsRouter.ts

```diff
@@ -243,6 +249,44 @@ export function createParticipantsRouter() {
     }
   );
 
+  participantsRouter.post(
+    '/:participantId/keyPair/add',
+    async (req: ParticipantRequest, res: Response) => {
+      const { participant } = req;
+      if (!participant?.siteId) {
+        return res.status(400).send('Site id is not set');
+      }
+      const { name, disabled } = keyPairParser.parse(req.body);
+      const currentUser = await findUserByEmail(req.auth?.payload?.email as string);
+      const auditTrail = await insertKeyPairAuditTrails(
+        participant,
+        currentUser!.id,
+        currentUser!.email,
+        AuditAction.Add,
+        name,
+        disabled
+      );
+
+      const keyPairs = await addKeyPair(participant.siteId, name, disabled);
+
+      await updateAuditTrailToProceed(auditTrail.id);
+      return res.status(201).json(keyPairs);
+    }
+  );
+
+  participantsRouter.get(
```

Comment:
nit: May I ask for the reason for moving this from site routes? For me, I feel as the participant route keeps growing, it makes sense to have those site-oriented APIs to be part of sites routes? (Key pair management, sharing permission management and domain management)

Reply:
The main reason is that I wanted this behind participant validation and didn't want to recreate that in another router.  The way I wrote this, the participant will only see their own sites (once that's done on the back end).  As it was implemented in the site router, anyone could craft a request to see another site's keys without any sort of permission checks.